### PR TITLE
pipeline switches in daemons, ml warehause as source of LIMs data

### PR DESCRIPTION
**Pull request by mgcam**

### CODE REVIEW

**Comment by mgcam** [comment]
As an added bonus, runs with batch unreleased (no data in ml iseq_warehouse flowcell table) will not go into analysis.

**Comment by mgcam** [comment]
Comments after a preliminary review by djk:
1. Using gclp glag as a pipeline option is not compatible with teh current analysis pipeline on the gclp cluster; we shoudl maintain compatibility.
2. create dry-run option for the pipeline daemons so that they can be tested against libe dbs.

File: lib/npg_pipeline/daemons/base.pm [hunk 1]
```diff
@@ -33,37 +29,61 @@ Readonly::Scalar my $NO_LIMS_LINK => -1;
...
+  isa        => q{Log::Log4perl::Logger},
+  is         => q{ro},
+  metaclass  => 'NoGetopt',
+  default    => sub { Log::Log4perl->get_logger() },
+);
+
+around 'log' => sub {
+   my $orig = shift;
+   my $self = shift;
+   $self->logger->info('"log" method is depricated');
```
Comment (dkj) [inline lib/npg_pipeline/daemons/base.pm]: deprecated

Comment (mgcam) [inline lib/npg_pipeline/daemons/base.pm]: > + $self->logger->info('"log" method is depricated');
>
> deprecated
>
Thank you, fixed.




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Pull request merged**